cmake/modules/FindZephyr-sdk.cmake

```diff
@@ -60,7 +60,44 @@ if(("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT}) OR
       set(ZEPHYR_TOOLCHAIN_VARIANT ${ZEPHYR_CURRENT_TOOLCHAIN_VARIANT})
     endif()
   else()
-    find_package(Zephyr-sdk ${Zephyr-sdk_FIND_VERSION} REQUIRED QUIET CONFIG PATHS
+    # Search for Zephyr SDK version 0.0.0 which does not exist, this is needed to
+    # return a list of compatible versions and find the best suited version that
+    # is available.
+    find_package(Zephyr-sdk 0.0.0 EXACT QUIET CONFIG PATHS
+                 /usr
+                 /usr/local
+                 /opt
+                 $ENV{HOME}
+                 $ENV{HOME}/.local
+                 $ENV{HOME}/.local/opt
+                 $ENV{HOME}/bin)
+
+    set(PACKAGE_FIND_VERSION ${Zephyr-sdk_FIND_VERSION})
+    set(COMPATIBLE_SDK_VERSION 0.0.0)
+
+    # Load the version file of each potential Zephyr SDK version and check if it
+    # is greater than the existing best found version.
+    foreach(ZEPHYR_SDK_CANDIDATE ${Zephyr-sdk_CONSIDERED_CONFIGS})
+      string(REGEX REPLACE "\.cmake$" "Version.cmake" ZEPHYR_SDK_VERSION_CANDIDATE ${ZEPHYR_SDK_CANDIDATE})
+
+      include(${ZEPHYR_SDK_VERSION_CANDIDATE} NO_POLICY_SCOPE)
+
+      if("${PACKAGE_VERSION_COMPATIBLE}" STREQUAL "TRUE")
```

Comment:
`PACKAGE_VERSION_COMPATIBLE` is a value which must be set by the package version file, as described:
> The version file must use these variables to check whether it is compatible or an exact match for the requested version and set the following variables with results:
> ...
> PACKAGE_VERSION_COMPATIBLE
>    True if version is compatible

so in principle this value value should be false, but because you manually loaded `Version` file and bypassed `find_package()` you have it available.
This makes this code harder to read and maintain.

Secondly, never do a CMake boolean test using a `STREQUAL`, cause the following values are also valid as `true`.
> True if the constant is 1, ON, YES, TRUE, Y, or a non-zero number (including floating point numbers)

Reply:
Fixed